src/transformStateWithClones.js

```diff
@@ -8,6 +9,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const table = {
+    ...state,
+  };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      for (const delet in table) {
+        delete table[delet];
+      }
+      console.log(table);
+    } else if (action.type === 'removeProperties') {
+      for (const remove of action.keysToRemove) {
+        delete table[remove];
+      }
+    } else {
+      Object.assign(table, action.extraData);
+    }
+
+    result.push({ ...table });
+  }
+
+  return result;
```

Comment:
Consider using a `switch` statement for handling different action types since you have a limited and known set of action types. This will make the code more readable and maintainable. For example: `switch (action.type) { case 'clear': ... break; case 'removeProperties': ... break; case 'addProperties': ... break; default: throw new Error('Invalid action type'); }`